src/modules/logcollector/src/logcollector.cpp

```diff
@@ -28,18 +29,29 @@ void Logcollector::EnqueueTask(boost::asio::awaitable<void> task) {
     boost::asio::co_spawn(m_ioContext, std::move(task), boost::asio::detached);
 }
 
-void Logcollector::Setup(const configuration::ConfigurationParser& configurationParser) {
-    m_enabled = configurationParser.GetConfig<bool>("logcollector", "enabled").value_or(config::logcollector::DEFAULT_ENABLED);
+void Logcollector::Setup(std::shared_ptr<const configuration::ConfigurationParser> configurationParser) {
+    if (!configurationParser) {
+        LogError("Invalid Configuration Parser passed to setup, module set to disabled.");
+        m_enabled = false;
+        return;
+    }
+
+    m_enabled = configurationParser->GetConfig<bool>("logcollector", "enabled").value_or(config::logcollector::DEFAULT_ENABLED);
+
+    if (m_ioContext.stopped())
```

Comment:
Why is this check necessary? Shouldn't be always stopped at this point?

Reply:
No, I was testing and when the module starts, in the first call to Setup() the state of `m_ioContext `is “running”.